s3torchconnector/tst/unit/test_s3_client.py

```diff
@@ -48,3 +52,29 @@ def test_list_objects_log(s3_client: S3Client, caplog):
     with caplog.at_level(logging.DEBUG):
         s3_client.list_objects(TEST_BUCKET, TEST_KEY)
     assert f"ListObjects {S3_URI}" in caplog.messages
+
+
+def test_s3_client_default_user_agent():
+    s3_client = S3Client(region=TEST_REGION)
+    expected_user_agent = f"s3torchconnector/{__version__}"
+    assert s3_client.user_agent_prefix == expected_user_agent
+    assert s3_client._client.user_agent_prefix == expected_user_agent
+
+
+def test_s3_client_custom_user_agent():
+    s3_client = S3Client(
+        region=TEST_REGION, user_agent=UserAgent(["component/version", "metadata"])
+    )
+    expected_user_agent = (
+        f"s3torchconnector/{__version__} (component/version; metadata)"
+    )
+    assert s3_client.user_agent_prefix == expected_user_agent
+    assert s3_client._client.user_agent_prefix == expected_user_agent
+
+
+@given(text())
+def test_user_agent_always_starts_with_package_version(comments):
```

Comment:
I don't think this is doing what you think it's doing :)
Perhaps `UserAgent` should assert it's given a list, not a string...